nixos/modules/services/misc/home-assistant.nix

```diff
@@ -238,20 +238,38 @@ in {
   config = mkIf cfg.enable {
     networking.firewall.allowedTCPPorts = mkIf cfg.openFirewall [ cfg.port ];
 
+    environment.etc = lib.mkMerge [
+      # symlink the configuration to /etc/home-assitant/configuration.nix
+      (lib.mkIf (cfg.config != null && !cfg.configWritable) {
+        "home-assistant/configuration.yaml".source = configFile;
+      })
+
+      (lib.mkIf (cfg.lovelaceConfig != null && !cfg.lovelaceConfigWritable) {
+        "home-assistant/ui-lovelace.yaml".source = lovelaceConfigFile;
+      })
+    ];
+
     systemd.services.home-assistant = {
       description = "Home Assistant";
       after = [ "network.target" ];
       preStart = optionalString (cfg.config != null) (if cfg.configWritable then ''
         cp --no-preserve=mode ${configFile} "${cfg.configDir}/configuration.yaml"
       '' else ''
         rm -f "${cfg.configDir}/configuration.yaml"
-        ln -s ${configFile} "${cfg.configDir}/configuration.yaml"
+        ln -s /etc/home-assistant/configuration.yaml "${cfg.configDir}/configuration.yaml"
       '') + optionalString (cfg.lovelaceConfig != null) (if cfg.lovelaceConfigWritable then ''
         cp --no-preserve=mode ${lovelaceConfigFile} "${cfg.configDir}/ui-lovelace.yaml"
       '' else ''
         rm -f "${cfg.configDir}/ui-lovelace.yaml"
-        ln -s ${lovelaceConfigFile} "${cfg.configDir}/ui-lovelace.yaml"
+        ln -s /etc/home-assitant/ui-lovelace.yaml "${cfg.configDir}/ui-lovelace.yaml"
       '');
+
+      restartTriggers = [
+        configFile
+        lovelaceConfigFile
+      ];
+      reloadIfChanged = true;
```

Comment:
Mhm, this should probably depend on `!configWritable && !lovelaceConfigWritable`?

Reply:
Additionally, you should add a note somewhere reminding users of the security implications here - "Please restart this service whenever a security update is released" or some such.

Reply:
Actually that is already happening and I'm testing that in the VM test. As soon as the binary changes the service restarts anyway.

Reply:
What? With `reloadIfChanged`? I'm pretty sure it never used to... That is great.

ping @dasJ who is really deep into the switching script right now.
ping @mweinelt who maybe thought this wasn't the case?

Reply:
Until #49528 is properly fixed there is actually a workaround we are using downstream. It's ugly but works and it doesn't add security implications.
Just add a second service that reloads the main service and give the reload service the restart triggers. Then don't give the main service `restartIfChanged=true`

I have included parts of our exim module as an example:
<details>
<summary>exim module</summary>

```nix
{
  systemd.services.exim-reload = {
    description = "Exim config reload";
    wantedBy = [ "multi-user.target" ];
    stopIfChanged = false;

    restartTriggers = [ config.environment.etc."exim.conf".source ];

    script = ''
      # Ensure the unit is already active
      ${config.systemd.package}/bin/systemctl is-active exim.service >/dev/null || exit 0
      ${config.systemd.package}/bin/systemctl reload-or-restart exim.service
    '';

    sandbox = 2;
    serviceConfig = {
      Type = "oneshot";
      RemainAfterExit = true;
      User = "exim";
      Group = "exim";
      SystemCallFilter = "@system-service";
    };

    apparmor = {
      enable = true;
      extraConfig = ''
        /run/dbus/system_bus_socket rw,
      '';
    };
  };

  security.polkit.extraConfig = ''
    polkit.addRule(function(action, subject) {
      if (action.id == "org.freedesktop.systemd1.manage-units" &&
        action.lookup("unit") == "exim.service" &&
        action.lookup("verb") == "reload-or-restart" &&
        subject.user == "exim") {
          return polkit.Result.YES;
      }
    });
  '';
}
```
</details>

Reply:
@dasJ can you explain why this works as expected then? Did you have a look at the test code that I added to this PR?

Reply:
I looked at the code now. I have really no idea why that works. Are you 100% sure it gets reloaded once and restarted the second time and it's not hass restarting itself? Printing the switch-to-configuration.pl call should explain what it's asking systemd to do

Reply:
Unfortunately the reloadIfChanged feature isn't really documented well (or correct). My current working theory is that the `restartIfChanged` list *is an additional list of restart triggers* and the `reloadIfChanged` boolean switches to reloading instead of restarting *when the list changes*. Meaning that it actually does exactly what we want in this case. The naming of these options is very unfortunate but I think it is working as expected.

I'll see if I can get this deployed tonight and give it some real-world testing as well was writing a VM test for the restartIfChanged semantics.

Reply:
ping @flokli because of past conversations about how we thought if `ExecStart` ever changes the service should be restarted... we were both under the impression `reloadIfChanged` forces the service to **never** restart. Brief conversations with @mweinelt leave me with the impression that they also think this... :thinking: 

Thanks for offering to look into this @andir!